docs/design/2022-07-28-support-generate-auto_random-pk.md

```diff
@@ -0,0 +1,100 @@
+# Support allocating auto_random primary key 
+
+- Author(s): [yangxin](http://github.com/xuanyu66)
+- Tracking Issue: https://github.com/tidb-incubator/TiBigData/issues/235
+
+## Table of Contents
+* [Real-time wide table](#real-time-wide-table)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Motivation or Background](#motivation-or-background)
+  * [Detailed Design](#detailed-design)
+    * [the mechanism of shard](#the-mechanism-of-shard)
+    * [Get the shardBits](#get-the-shardbits)
+    * [Generate the currentShard with startTimestamp](#generate-the-currentshard-with-starttimestamp)
+    * [Use IGNORE_AUTO_RANDOM_COLUMN_VALUE to ignore the explicit primary key.](#use-ignore_auto_random_column_value-to-ignore-the-explicit-primary-key)
+  * [Compatibility](#compatibility)
+  * [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+
+## Introduction
+
+This design doc is about how to support allocating auto_random primary key in flink-tidb-connector.
+
+## Motivation or Background
+
+[AUTO_RANDOM](https://docs.pingcap.com/tidb/dev/auto-random) is a feature introduced by TiDB in order to increase the performance of the insert operation 
+when the writing hotspot issue occurs caused by auto-increment.
+
+The primary key, which must be bigint type and clustered, is allocated in the following way:
+The highest n(default is five，1 <= n < 16) digits (ignoring the sign bit) of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. 
+The remaining digits are allocated values in an auto-increment order.
+
+Luckily, we already support auto_increment and [shard_row_id_bits](https://docs.pingcap.com/tidb/dev/shard-row-id-bits) in Flink,
+of which the mechanism of the latter is similar to the auto_random, so we can reuse the methods.
+
+## Detailed Design
+
+Firstly, we will introduce the mechanism of shard and then discuss how wo get shardBits and get the shardSeed with good collision resistance.
+
+### the mechanism of shard
+
+Firstly, we need to generate the shardSeed, whose bits' length must be greater than and equal to 16.
```

Comment:
```suggestion
Firstly, we need to generate the shardSeed, whose bits' length must be greater than or equal to 16.
```
shardBits can't be 16 right? so why shardSeed should >=16 rather than >=15?